lib/active_storage/service/mirror_service.rb

```diff
@@ -18,24 +18,24 @@ def initialize(primary:, mirrors:)
   end
 
   def upload(key, io, checksum: nil)
-    each_service.collect do |service|
-      service.upload key, io.tap(&:rewind), checksum: checksum
+    perform_async_across_services do |service|
+      service.upload key, StringIO.new(io.tap(&:rewind).read), checksum: checksum
```

Comment:
I'm almost sure that this is not the right way to create a new IO instance but `clone` or `dup` don't work. When `read` is called it mutates all instances.